assets/chef-recipes/cookbooks/mariadb/recipes/uninstall.rb

```diff
@@ -1,19 +1,49 @@
+service 'mariadb' do
+  action :stop
+end
+
+execute 'Reset iptables settings' do
+  command 'iptables -D INPUT -p tcp --dport 3306 -j ACCEPT -m state --state NEW'
+end
+
 case node[:platform_family]
-when "debian"
-  package "mariadb-common" do
+when 'debian'
+  package 'mariadb-common' do
+    action :purge
+  end
+  file '/etc/apt/sources.list.d/mariadb.list' do
+    action :delete
+  end
+  apt_update 'update apt cache' do
+    action :update
+  end
+  execute 'Save iptables settings' do
+    command 'iptables-save > /etc/iptables/rules.v4'
+  end
+
+when 'rhel', 'centos'
+  package 'MariaDB-common' do
     action :remove
   end
-  execute "Remove mariadb repository" do
-    command "rm -fr /etc/apt/sources.list.d/mariadb.list"
+  file '/etc/yum.repos.d/mariadb.repo' do
```

Comment:
The same here: [`yum_repository`](https://docs.chef.io/resources/yum_repository/#examples) has the `:delete` action.